### PR TITLE
Fix ifp->eport init

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

File: mongoose.c [hunk 1]
```diff
@@ -7907,8 +7908,9 @@ void mg_tcpip_init(struct mg_mgr *mgr, struct mg_tcpip_if *ifp) {
     mgr->extraconnsize = sizeof(struct connstate);
     if (ifp->ip == 0) ifp->enable_dhcp_client = true;
     memset(ifp->gwmac, 255, sizeof(ifp->gwmac));  // Set to broadcast
-    mg_random(&ifp->eport, sizeof(ifp->eport));   // Randomise the initial
-    ifp->eport += MG_EPHEMERAL_PORT_BASE;         // ephemeral port
+    mg_random(&ifp->eport, sizeof(ifp->eport));   // Random from 0 to 65536
+    ifp->eport |=
+        MG_EPHEMERAL_PORT_BASE;  // Random from MG_EPHEMERAL_PORT_BASE to 65536
```
Comment (scaprile) [inline mongoose.c:7913]: 0 to 65535 and MG_EPHEMERAL_PORT_BASE to 65535

File: src/tcpip/tcpip.c [hunk 2]
```diff
@@ -856,8 +857,9 @@ void mg_tcpip_init(struct mg_mgr *mgr, struct mg_tcpip_if *ifp) {
     mgr->extraconnsize = sizeof(struct connstate);
     if (ifp->ip == 0) ifp->enable_dhcp_client = true;
     memset(ifp->gwmac, 255, sizeof(ifp->gwmac));  // Set to broadcast
-    mg_random(&ifp->eport, sizeof(ifp->eport));   // Randomise the initial
-    ifp->eport += MG_EPHEMERAL_PORT_BASE;         // ephemeral port
+    mg_random(&ifp->eport, sizeof(ifp->eport));   // Random from 0 to 65536
+    ifp->eport |=
+        MG_EPHEMERAL_PORT_BASE;  // Random from MG_EPHEMERAL_PORT_BASE to 65536
```
Comment (scaprile) [inline src/tcpip/tcpip.c:862]: 0 to 65535 and MG_EPHEMERAL_PORT_BASE to 65535

File: src/tcpip/tcpip.c [hunk 3]
```diff
@@ -759,6 +759,7 @@ static void mg_tcpip_rx(struct mg_tcpip_if *ifp, void *buf, size_t len) {
     rx_ip(ifp, &pkt);
   } else {
     MG_DEBUG(("  Unknown eth type %x", mg_htons(pkt.eth->type)));
+    mg_hexdump(buf, len >= 16 ? 16 : len);
```
Comment (scaprile) [inline src/tcpip/tcpip.c:762]: Maybe we could let VLANs pass, ethtype = 6; we know them, we just don't support them (yet ?)

Comment (cpq) [reply src/tcpip/tcpip.c:762]: I believe you're correct!

File: src/tcpip/tcpip.c (hunk 2 above)
Comment (cpq) [reply src/tcpip/tcpip.c:862]: Done

**Review by scaprile — APPROVED** [review]

**Pull request merged**